poisson_lib.py

```
@@ -128,647 +163,625 @@ def analyze_poisson_outcome(
-                                coef = params[d_name]
+                            d_name = f"{c}::{lvl}"
+                            multi_df[d_name] = (raw_vals.astype(str) == str(lvl)).astype(int)
+                else:
+                    multi_df[c] = df_aligned[c].apply(clean_numeric_value)
+            
+            # ✅ Add interaction terms if specified
+            if interaction_pairs:
+                try:
+                    from interaction_lib import create_interaction_terms, format_interaction_results
+                    multi_df, int_meta = create_interaction_terms(multi_df, interaction_pairs, mode_map)
+                    logger.info(f"✅ Added {len(int_meta)} interaction terms to Poisson multivariate model")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)